### ISSUE TITLE
Interface hangs when game fails to download

### ISSUE THREAD

**Issue by JaapJoris** [rtts/ialauncher#9]
Downloading a game can fail for a number of reasons:

- No internet connection
- Download interrupted
- Invalid server response
- The file was removed from the Internet Archive

Particularly the last reason has cropped up: several games were taken offline by the Internet Archive since IA Launcher was released.

Currently, when the download background process fails the user interface just hangs, because it assumes it will always move to a playing state after the downloading state. The code needs to be refactored to be able to handle failed downloads. Also, games that once existed on the Internet Archive but were taken offline seem to successfully download a file of 0 bytes, which crashes the unzipping method.

In summary, better error handling is needed.
